keps/sig-auth/4940-psa-block-host-field-in-probes/README.md

```diff
@@ -0,0 +1,354 @@
+
+* container.LivenessProbe.ProbeHandler.HTTPGet.Host
+* container.ReadinessProbe.ProbeHandler.HTTPGet.Host
+* container.StartupProbe.ProbeHandler.HTTPGet.Host
+* container.LivenessProbe.ProbeHandler.TCPSocket.Host
+* container.ReadinessProbe.ProbeHandler.TCPSocket.Host
+* container.StartupProbe.ProbeHandler.TCPSocket.Host
+* container.Lifecycle.PostStart.TCPSocket.Host // Deprecated. TCPSocket is NOT supported as a LifecycleHandler and kept for backward compatibility.
+* container.Lifecycle.PreStop.TCPSocket.Host // Deprecated. TCPSocket is NOT supported as a LifecycleHandler and kept for backward compatibility.
+* container.Lifecycle.PostStart.HTTPGet.Host
+* container.Lifecycle.PreStop.HTTPGet.Host
+
+These above exact set of fields is what will get deprecated as well.
+
+### Test Plan
+
+* Unit and E2E tests will be added to ensure the PSA works as expected
+
+##### Prerequisite testing updates
+
+None
+
+##### Unit tests
+
+Necessary unit tests will be added to the [PSA package] for
+testing the new code.
+Current test coverage status for the package is:
+- `k8s.io/pod-security-admission/policy`: `2025-05-06` - `89.9%`
+- `k8s.io/pod-security-admission/test`: `TBD` - `TBD`
+
+[PSA package]: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/pod-security-admission/policy
+
+##### Integration tests
+
+The following integration tests will be added to verify the PSA validation logic:
+
+1. Test that pods with `.host` field set in probes are rejected when PSA is enabled with baseline level
+2. Test that pods without `.host` field set in probes are allowed when PSA is enabled with baseline level
+3. Test that existing pods with `.host` field set continue to work when PSA is enabled
+4. Test that pods with `.host` field set are allowed when PSA is disabled or using an older version
+
+These tests will be added to:
+- `test/integration/auth/podsecurity_test.go`
+https://storage.googleapis.com/k8s-triage/index.html?test=TestPodSecurity
+
+The integration tests will verify the PSA policy validation logic by:
+- Creating test cases for each probe type (HTTPGet, TCPSocket)
+- Testing each probe location (LivenessProbe, ReadinessProbe, StartupProbe, Lifecycle)
+- Verifying the PSA policy enforcement at the baseline level
+- Testing the behavior with different PSA configurations
+
+##### e2e tests
+
+There are no Pod Security specific E2E tests (we rely on integration test coverage instead), but the Pod Security admission controller is enabled in E2E clusters, and all E2E test namespaces are labeled with the enforcement label for Pod Security.
+
+### Graduation Criteria
+
+The PSA added will be done within a single release
+and given there will be no feature gates for that,
+there is no need for multi-release graduation criteria.
+All related code will land within the same single release
+
+
+#### Deprecation
+
+`.Host` field will be deprecated from the Probe and Lifecycle Handlers
```

Comment:
Hm... maybe remove the `Deprecation` heading and just include this as part of `Graduation Criteria`.

Since the deprecation consists _only_ of marking the field deprecated, and there is no intention of removing it or introducing any replacement for it, this is also something that can happen in a single release.